mesonbuild/mesonlib.py

```diff
@@ -1391,3 +1391,58 @@ class LibType(Enum):
     STATIC = 1
     PREFER_SHARED = 2
     PREFER_STATIC = 3
+
+
+class ProgressBar:
+    """Wrapper class for tqdm"""
+    def __init__(self, iterable=None, total=None, bar_type=None, desc=None):
+        if iterable is not None:
+            self.iterable = iter(iterable)
+            return
+        self.total = total
+        self.done = 0
+        self.printed_dots = 0
+        if self.total and bar_type == 'download':
+            print('Download size:', self.total)
+        if desc:
+            print('{}: '.format(desc), end='')
+
+    # Pretend to be an iterator when called as one and don't print any
+    # progress
+    def __iter__(self):
+        return self.iterable
+
+    def __next__(self):
+        return next(self.iterable)
+
+    def print_dot(self):
+        print('.', end='')
+        sys.stdout.flush()
+        self.printed_dots += 1
+
+    def update(self, progress):
+        self.done += progress
+        if not self.total:
+            # Just print one dot per call if we don't have a total length
+            self.print_dot()
+            return
+        ratio = int(self.done / self.total * 10)
+        while self.printed_dots < ratio:
+            self.print_dot()
+
+    def close(self):
+        print('')
+
+try:
+    raise ImportError
```

Comment:
Forgotten debug code?

If yes and this needs fixing then also consider switching the names around? That is, provide and use a `ProgressBar` class in our code that may (or may not) be implemented with `tqdm`. The latter is a confusing name to encounter out of the blue.

Reply:
Yes, that's debug code. Good point re: implementing it via the class. I'll make it happen.